solr/solrj/src/java/org/apache/solr/common/util/ExecutorUtil.java

```diff
@@ -196,7 +198,13 @@ public void execute(final Runnable command) {
 
       String ctxStr = contextString.toString().replace("/", "//");
       final String submitterContextStr = ctxStr.length() <= MAX_THREAD_NAME_LEN ? ctxStr : ctxStr.substring(0, MAX_THREAD_NAME_LEN);
-      final Exception submitterStackTrace = enableSubmitterStackTrace ? new Exception("Submitter stack trace") : null;
+      final Exception submitterStackTrace;
+      if (enableSubmitterStackTrace) {
+        Exception grandParentSubmitter = submitter.get();
+        submitterStackTrace = new Exception("Submitter stack trace", grandParentSubmitter);
```

Comment:
Hmm; isn't this one of those cases we should call fillInStackTrace() on the exception because we don't throw it?

Reply:
I don't understand the suggestion - Throwable's constructor already calls it for us.

Reply:
Ah nevermind 👍